FAIMS/FAIMS3
# Customise Deployment of Android App

# Customise Deployment of Android App

## JIRA Ticket

[BSS-405](https://jira.csiro.com/browse/BSS-405)
[BSS-344](https://jira.csiro.com/browse/BSS-344)

## Description

Makes a few more things customisable in the app build to facilitate deployment to the Google Play Store as a separate app. 

## Proposed Changes

- Adds environment variables for some previously fixed values relating to app store deployment
- Adds documentation of the Android deployment workflow for future reference
- Automatically generate app icons based on configured theme
- Automatically update app name and id based on configured values

Note. tested on Android deployment but also touches some parts of Apple deployment which will need testing and fixing later.

## How to Test

Hard to test independantly but nothing here should break the build. Note that there is a new task run before starting the app to generate icons so one immediate change you see will be the main app icon will change with theme changes (need to re-run `npm start` since it is a pre-start step). 

## Checklist
